src/debug_utils.cc

```diff
@@ -100,35 +100,104 @@ class Win32SymbolDebuggingContext final : public NativeSymbolDebuggingContext {
+
+    pSymbol->SizeOfStruct = sizeof(SYMBOL_INFO);
+    pSymbol->MaxNameLen = MAX_SYM_NAME;
+
+    if (SymFromAddr(current_process_, dwAddress, &dwDisplacement, pSymbol)) {
+      // SymFromAddr returned success
+      return NameAndDisplacement(pSymbol->Name, dwDisplacement);
+    } else {
+      // SymFromAddr failed
+      const DWORD error = GetLastError();  // "eat" the error anyway
+#ifdef DEBUG
+      fprintf(stderr, "SymFromAddr returned error : %lu\n", error);
+#endif
+    }
+    // End MSDN code
 
-    info->MaxNameLen = MAX_SYM_NAME;
-    info->SizeOfStruct = sizeof(SYMBOL_INFO);
+    return NameAndDisplacement();
+  }
 
-    SymbolInfo ret;
-    const bool have_info = SymFromAddr(current_process_,
-                                       reinterpret_cast<DWORD64>(address),
-                                       nullptr,
-                                       info);
-    if (have_info && strlen(info->Name) == 0) {
-      if (UnDecorateSymbolName(info->Name,
-                               demangled,
-                               sizeof(demangled),
-                               UNDNAME_COMPLETE)) {
-        ret.name = demangled;
-      } else {
-        ret.name = info->Name;
-      }
+  SymbolInfo WrappedGetLine(DWORD64 dwAddress) const {
+    SymbolInfo sym{};
+
+    // Refs: https://docs.microsoft.com/en-us/windows/desktop/Debug/retrieving-symbol-information-by-address
+    // Patches:
+    // Use `fprintf(stderr, ` instead of `printf`.
+    // Assign values to `sym` on success.
+    // `current_process_` instead of `hProcess.
+
+    // Patch: made into arg - DWORD64  dwAddress;
+    DWORD dwDisplacement;
+    IMAGEHLP_LINE64 line;
+
+    SymSetOptions(SYMOPT_LOAD_LINES);
+
+    line.SizeOfStruct = sizeof(IMAGEHLP_LINE64);
+    // Patch: made into arg - dwAddress = 0x1000000;
+
+    if (SymGetLineFromAddr64(current_process_, dwAddress,
+                             &dwDisplacement, &line)) {
+      // SymGetLineFromAddr64 returned success
+      sym.filename = line.FileName;
+      sym.line = line.LineNumber;
+    } else {
+      // SymGetLineFromAddr64 failed
+      const DWORD error = GetLastError();  // "eat" the error anyway
+#ifdef DEBUG
+      fprintf(stderr, "SymGetLineFromAddr64 returned error : %lu\n", error);
+#endif
+    }
+    // End MSDN code
+
+    return sym;
+  }
+
+  // Fills the SymbolInfo::name of the io/out argument `sym`
+  std::string WrappedUnDecorateSymbolName(const char* name) const {
+    // Refs: https://docs.microsoft.com/en-us/windows/desktop/Debug/retrieving-undecorated-symbol-names
+    // Patches:
+    // Use `fprintf(stderr, ` instead of `printf`.
+    // return `szUndName` instead of `printf` on success
+    char szUndName[MAX_SYM_NAME];
+    if (UnDecorateSymbolName(name, szUndName, sizeof(szUndName),
+                             UNDNAME_COMPLETE)) {
+      // UnDecorateSymbolName returned success
+      return szUndName;
+    } else {
+      // UnDecorateSymbolName failed
```

Comment:
I am not a fan of this type of comment: What happens is obvious in both branches, and the branches themselves are short. (And I think we have a convention of ending comments with a period.)

Reply:
This code was lifter verbatim from https://docs.microsoft.com/en-us/windows/desktop/Debug/retrieving-symbol-information-by-address
I would rather keep changes to minimon, for future reverse engineering.